src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantCanonicalRepresentation.java

```diff
@@ -100,6 +98,31 @@
     private final List<String> eventDescriptions;
     private final byte[] altSeq;
 
+    @VisibleForTesting
+    CpxVariantCanonicalRepresentation(final SimpleInterval affectedRefRegion,
+                                      final List<SimpleInterval> referenceSegments,
+                                      final List<String> eventDescriptions,
+                                      final byte[] altSeq) {
+        this.affectedRefRegion = affectedRefRegion;
+        this.referenceSegments = referenceSegments;
+        this.eventDescriptions = eventDescriptions;
+        this.altSeq = altSeq;
+    }
+
+    @VisibleForTesting
+    static CpxVariantCanonicalRepresentation makeInterpretation(final CpxVariantInducingAssemblyContig cpxVariantInducingAssemblyContig) {
+
+        if (cpxVariantInducingAssemblyContig.getEventPrimaryChromosomeSegmentingLocations().size() == 1) {
```

Comment:
Why have two different code paths here and two different constructors? All this method is doing is looking at data in the `cpxVariantInducingAssemblyContig`, so why not just have one constructor that takes that? Then this method can be super simple or just go away.

Reply:
changed as suggested. now much cleaner. Thanks!